packages/components/src/components/flyout/flyout-styles.ts

```diff
@@ -0,0 +1,144 @@
+import { getCss, isThemeDark, Theme } from '../../utils';
+import {
+  addImportantToEachRule,
+  getFrostedGlassBackgroundJssStyles,
+  getInsetJssStyle,
+  getStickyJSSStyle,
+  getThemedColors,
+  hostHiddenStyles,
+  hoverMediaQuery,
+} from '../../styles';
+import { FLYOUT_Z_INDEX } from '../../constants';
+import { spacingFluidLarge, spacingStaticMedium } from '../../../../utilities/projects/utilities';
+import type { FlyoutPosition } from './flyout-utils';
+
+export const flyoutBoxShadowColor = 'rgba(204, 204, 204, 0.35)';
+export const flyoutBoxShadowColorDark = 'rgba(0, 0, 0, 0.6)';
+
+const flyoutTransitionDuration = '0.5s';
+const flyoutTransitionTimingFunction = 'cubic-bezier(0.77, 0, 0.175, 1)';
+
+const cssVariableMaxWidth = '--p-flyout-max-width';
+const maxWidthDefault = '1180px';
+const minWidthDefault = '320px';
+
+export const getComponentCss = (
+  isOpen: boolean,
+  position: FlyoutPosition,
+  hasHeader: boolean,
+  hasFooter: boolean,
+  hasSubFooter: boolean,
```

Comment:
Can there be a subFooter without a footer or is there a relation between them? Is/should this be validated somewhere?